content/blog/tips-for-faster-ci-builds/index.md

```diff
@@ -0,0 +1,218 @@
+
+{% info(title="Want a Real-World Example?", icon="crab") %}
+
+Check out this production-ready GitHub Actions workflow that implements all the tips from this article:
+[click here](https://github.com/lycheeverse/lychee/blob/master/.github/workflows/ci.yml).
+
+{% end %}
+
+## Use Swatinem's cache action
+
+This is easily my most important recommendation on this list.
+
+My friend [Arpad Borsos](https://github.com/swatinem), also known as Swatinem, has created a cache action specifically tailored for Rust projects.
+It's an excellent way to speed up any Rust CI build and requires no code changes to your project.
+
+```yaml
+name: CI
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest-arm64
+
+    env:
+      CARGO_TERM_COLOR: always
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+
+      # The secret sauce!
+      - uses: Swatinem/rust-cache@v2
+
+      - run: |
+          cargo check
+          cargo test
+          cargo build --release
+```
+
+The action requires no additional configuration and works out of the box.
+There's no need for a separate step to store the cache — this happens automatically through a post-action.
+This approach ensures that broken builds aren't cached, and for subsequent builds, you can save several minutes of build time.
+
+Here's the [documentation](https://github.com/Swatinem/rust-cache) where you can learn more. 
+
+## Use the `--locked` flag
+
+When running `cargo build`, `cargo test`, or `cargo check`, you can pass the `--locked` flag to prevent Cargo from updating the `Cargo.lock` file.
+
+This is particularly useful for CI operations since you save the time to update dependencies.
+Typically you want to test the exact dependency versions specified in your lock file anyway.
+
+```yaml
+- run: cargo check --locked
+- run: cargo test --locked
+```
+
+## Use the new ARM64 runners
+
+GitHub Actions has recently announced that
+Linux ARM64 hosted runners are now available for free in public repositories. 
+[Here's](https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/) the announcement.
+
+Switching to ARM64 provides up to 40% performance improvement and is straightforward. Simply replace `ubuntu-latest` with `ubuntu-latest-arm64` in your workflow file:
+
+```yaml
+jobs:
+  test:
+    runs-on: ubuntu-latest-arm64
+```
+
+However, in my [tests](https://github.com/lycheeverse/lychee/pull/1614), the downside was that it took a long time until a runner was allocated to the job. The waiting time dwarfed the actual build time. I assume GitHub will add more runners in the future to mitigate this issue.
+
+## Use `cargo-chef` For Docker Builds 
+
+For Rust Docker images, [`cargo-chef`](https://github.com/LukeMathWalker/cargo-chef) can significantly speed up the build process by leveraging Docker's layer caching:
```

Comment:
You could also point out that all of this is also possible by handcrafting your Dockerfile to use cache mounts and copying only Cargo.* and creating an empty src/main.rs, but that this is way easier.

Solid recommendation, I really like it :+1:

Reply:
added. Not sure if we want to keep the full example